src/charm.py

```diff
@@ -979,6 +1057,118 @@ def auth_enabled(self) -> bool:
 
         return any("MONGOD_ARGS" in line and "--auth" in line for line in env_vars)
 
+    def _scope_obj(self, scope: Scopes):
+        if scope == APP_SCOPE:
+            return self.app
+        if scope == UNIT_SCOPE:
+            return self.unit
+
+    def _peer_data(self, scope: Scopes):
+        if not self._peers:
+            return {}.setdefault(scope, {})
+        scope_obj = self._scope_obj(scope)
+        return self._peers.data[scope_obj]
+
+    def _juju_secret_set(self, scope: Scopes, key: str, value: str) -> str:
+        """Helper function setting Juju secret."""
+        peer_data = self._peer_data(scope)
+        self._juju_secrets_get(scope)
+
+        secret = self.secrets[scope].get(Config.Secrets.SECRET_LABEL)
+
+        # It's not the first secret for the scope, we can re-use the existing one
+        # that was fetched in the previous call
```

Comment:
good comment, can you add a few words on why this is important, Maybe something like:
```
        # It's not the first secret for the scope, we can re-use the existing one
        # that was fetched in the previous call, as fetching secrets from juju is
        # slow
``` 
(not sure what the reason is, but that is a guess)

Reply:
This code is actually from k8s charm with implementation of secrets, but I will update the comment, thank you 😄